### DISCUSSION TITLE
Support connection to AWS Bedrock for LLMs

### DISCUSSION THREAD

**Discussion by mdlmarkham** [open-webui/open-webui#5148] [Ideas]
While I'm running Ollama for most of my models, some of the bigger ones are a challenge. It would be very nice to be able to connect directly to AWS Bedrock.

**Comment by juan-abia** [comment]
This would be awesome!

**Comment by bearlike** [comment]
Check out [Lite-LLM](https://github.com/BerriAI/litellm). It support proxying models from different provider. I proxy Bedrock model via it.

**Comment by kevingoad-arcfield** [comment]
We have this running w/ no issues:
https://github.com/aws-samples/bedrock-access-gateway

**Reply by pbpippen** [reply]
Same i have this setup as well, but on occasion do you get a network error model not found? And then have to hit generate a few times to get a response?

**Reply by kevingoad-arcfield** [reply]
No issues here.  We have bedrock access gateway running on Fargate sitting behind an internal ALB.

**Reply by pbpippen** [reply]
Mine is on ec2. Feel like it's something I'm overlooking.

**Reply by pbpippen** [reply]
Do you have image attachment issues when using bedrock access gateway? Mine only works with text.

**Reply by kevingoad-arcfield** [reply]
No issues when using vision-supported models (Sonnet & Haiku)

**Reply by codefox1387** [reply]
I also have Bedrock Access Gateway and it works fine for just the LLMs. What it doesn't work well for is accessing the knowledge bases, and RAG with knowledge bases. You need to pass auth, and receive back citations and context, but no support for that yet.

**Comment by jweisman11** [comment]
We also have this stack running and can share a few helpful tips and tricks we've learned along the way. 

QUESTION - Any concerns from the community if I contribute a tutorial to be added in OIU'S docs under the integrations section here: https://docs.openwebui.com/category/-integrations  Would be my first PR so checking first before writing and submitting.

- Open-WebUI
- AWS' Bedrock Access Gateway (BAG) - as the "proxy" to translate Bedrock's endpoints to conform to OpenAI schema required by OIU
- Amazon Bedrock's Foundation Models (FM) including Anthropic, Meta, Titan, etc.
